cluster-diagnosis/utils.py

```diff
@@ -131,6 +132,87 @@ def get_pod_config(pod_name):
     return output
 
 
+def get_pods_summarized_status_iterator(pod_name_substring, must_exist=True):
+    """Returns a summarized status of the pods by retrieving the status
+    multiple times.
+
+    This helps avoid any false negatives that can occur
+    in the scenario wherein the status is checked just after a pod restart.
+
+    Args:
+        pod_name_substring - the substring containing the pod name.
+        must_exist - boolean to indicate that a pod with that name must exist.
+            If the condition isn't satisfied, an error will be logged.
+
+    Returns:
+        name (string): name of the pod.
+        ready_status (string): the ready status of the pod.
+        status (string): the status of the pod (e.g. Running).
+        node_name (string): the name of the node.
+    """
+    pod_name_status_map = {}
+    for name, ready_status, status, node_name in \
+            get_pods_status_iterator(pod_name_substring, must_exist):
+        pod_name_status_map[name] = status
+        for attempt in range(0, 10):
+            # These retry attempts will take some time. Provide some form of
+            # visual feedback to the user.
+            # Cannot use log as it'll print on a new line every time.
+            sys.stdout.write('.')
+            sys.stdout.flush()
+            time.sleep(2)
```

Comment:
That’s a long sleep! I would exchange the loops, so that you could check other pods while waiting for one pod to come up. The outer loop would loop and wait 2 seconds as long as there’s at least one pod not up.

Reply:
And don’t sleep on the first iteration.

Reply:
Done.